snorkel/analysis/error_analysis.py

```diff
@@ -55,3 +56,59 @@ def get_label_buckets(*y: np.ndarray) -> Dict[Tuple[int, ...], np.ndarray]:
     for i, labels in enumerate(zip(*y_flat)):
         buckets[labels].append(i)
     return {k: np.array(v) for k, v in buckets.items()}
+
+
+def get_label_instances(
+    bucket: Tuple[int, ...], x: np.ndarray, *y: np.ndarray
+) -> np.ndarray:
+    """Return instances in x that were labeled according to bucket.
+
+    Parameters
+    ----------
+    bucket
+        A tuple of label values corresponding to which instances from x are returned
+    x
+        NumPy array of data instances to be returned
+    *y
+        A list of np.ndarray of (int) labels
+
+    Returns
+    -------
+    np.ndarray
+        NumPy array of instances that were labeled according to bucket
+
+    Example
+    -------
+    A common use case is calling ``get_label_instances(bucket, x.to_numpy(), Y_gold, Y_pred)`` where
+    ``x`` is a NumPy array of data instances used to generate labels, `Y_gold`` is a set of
+    gold (i.e. ground truth) labels, and ``Y_pred`` is a corresponding set of predicted labels.
+
+    >>> import pandas as pd
+    >>> x = pd.DataFrame(data={'col1': ["this is a string", "a second string", "a third string"], 'col2': ["1", "2", "3"]})
+    >>> Y_gold = np.array([1, 1, 1])
+    >>> Y_pred = np.array([1, 0, 0])
+    >>> bucket = (1, 0)
+
+    >>> get_label_instances(bucket, x.to_numpy(), Y_gold, Y_pred)
+    array([['a second string', '2'],
+           ['a third string', '3']], dtype=object)
+
+    The returned NumPy array of data instances corresponds to the data instances in x where
```

Comment:
As you know from your tests, the method isn't limited to just two label vectors. Let's make this comment more general or make it clear that it's talking specifically about the example above, perhaps by moving up by the other Example text, above the code.